server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreIT.java

```diff
@@ -97,76 +102,404 @@ private Map<String, Long> indexData(int numberOfIterations, boolean invokeFlush)
+    /**
+     * Simulates all data restored using Remote Translog Store.
+     * @throws IOException IO Exception.
+     */
     public void testRemoteTranslogRestoreWithNoDataPostCommit() throws IOException {
         testRestoreFlow(true, 1, true);
     }
 
+    /**
+     * Simulates all data restored using Remote Translog Store.
+     * @throws IOException IO Exception.
+     */
     public void testRemoteTranslogRestoreWithNoDataPostRefresh() throws IOException {
         testRestoreFlow(true, 1, false);
     }
 
+    /**
+     * Simulates refreshed data restored using Remote Segment Store
+     * and unrefreshed data restored using Remote Translog Store.
+     * @throws IOException IO Exception.
+     */
     public void testRemoteTranslogRestoreWithRefreshedData() throws IOException {
         testRestoreFlow(true, randomIntBetween(2, 5), false);
     }
 
+    /**
+     * Simulates refreshed data restored using Remote Segment Store
+     * and unrefreshed data restored using Remote Translog Store.
+     * @throws IOException IO Exception.
+     */
     public void testRemoteTranslogRestoreWithCommittedData() throws IOException {
         testRestoreFlow(true, randomIntBetween(2, 5), true);
     }
 
+    /**
+     * Simulates full data loss due to unrefreshed data, with no data restored from Remote Segment Store.
+     * @throws IOException IO Exception.
+     */
+    @AwaitsFix(bugUrl = "https://github.com/opensearch-project/OpenSearch/issues/8479")
+    public void testRSSRestoreWithNoDataPostCommitPrimaryReplicaDown() throws IOException {
+        testRestoreFlowBothPrimaryReplicasDown(false, 1, true);
+    }
+
+    /**
+     * Simulates full data loss due to unrefreshed data, with no data restored from Remote Segment Store.
+     * @throws IOException IO Exception.
+     */
+    @AwaitsFix(bugUrl = "https://github.com/opensearch-project/OpenSearch/issues/8479")
+    public void testRSSRestoreWithNoDataPostRefreshPrimaryReplicaDown() throws IOException {
+        testRestoreFlowBothPrimaryReplicasDown(false, 1, false);
+    }
+
+    /**
+     * Simulates data restored until the refreshed data in Remote Segment Store
+     * and data loss for the unrefreshed data.
+     * @throws IOException IO Exception.
+     */
+    @AwaitsFix(bugUrl = "https://github.com/opensearch-project/OpenSearch/issues/8479")
+    public void testRSSRestoreWithRefreshedDataPrimaryReplicaDown() throws IOException {
+        testRestoreFlowBothPrimaryReplicasDown(false, randomIntBetween(2, 5), false);
+    }
+
+    /**
+     * Simulates data restored until the refreshed data in Remote Segment Store
+     * and data loss for the unrefreshed data.
+     * @throws IOException IO Exception.
+     */
+    @AwaitsFix(bugUrl = "https://github.com/opensearch-project/OpenSearch/issues/8479")
+    public void testRSSRestoreWithCommittedDataPrimaryReplicaDown() throws IOException {
+        testRestoreFlowBothPrimaryReplicasDown(false, randomIntBetween(2, 5), true);
+    }
```

Comment:
lets omit it, in favor of both remote segment and translog enabled

Reply:
Just to confirm, if we omit RSS-only tests, we won't be able to assert on the exact data that is lost (in cases where there is unrefreshed data indexed before node failure). Is that okay? 
This is applicable to the above comment for the existing tests too.